iphone/Maps/Core/iCloud/DefaultLocalDirectoryMonitor.swift

```diff
@@ -63,10 +63,11 @@ final class DefaultLocalDirectoryMonitor: LocalDirectoryMonitor {
     guard state != .started else { return }
 
     let nowTimer = Timer.scheduledTimer(withTimeInterval: .zero, repeats: false) { [weak self] _ in
```

Comment:
Why a timer is needed?

Reply:
The timer is used to debounce to frequent local directory updates that can be triggered by the multiple writings

Reply:
Does interval .zero properly handle all changed files (e.g. when all lists are hidden)? Or only some files/changes are handled?

Reply:
Actually, this call is used to trigger the initial directory scanning (because the dispatch source will start working only for the next change and doesn't return the initial state).

`.zero` is used to trigger the scanning without delay (but after the method exits the scope - this task will be added to the main runloop so the .zero doesn't equal to 0).

The time when the foundation's method scans the directory should works well. The Apple says that `fileManager` is thread-safe and it should wait until the system finishes writing to the file.

Reply:
Why timer and not, for example, DispatchQueue.current.async?